kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/zookeeper/ZooKeeperInstanceClient.scala

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.ha.client.zookeeper
+
+import java.io.IOException
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicBoolean
+
+import org.apache.curator.framework.CuratorFramework
+import org.apache.curator.framework.recipes.nodes.PersistentNode
+import org.apache.curator.framework.state.ConnectionState
+import org.apache.curator.framework.state.ConnectionState.CONNECTED
+import org.apache.curator.framework.state.ConnectionState.LOST
+import org.apache.curator.framework.state.ConnectionState.RECONNECTED
+import org.apache.curator.framework.state.ConnectionStateListener
+import org.apache.zookeeper.WatchedEvent
+import org.apache.zookeeper.Watcher
+
+import org.apache.kyuubi.{KyuubiException, Logging}
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.ha.HighAvailabilityConf.HA_ZK_CONN_MAX_RETRIES
+import org.apache.kyuubi.ha.HighAvailabilityConf.HA_ZK_CONN_MAX_RETRY_WAIT
+import org.apache.kyuubi.ha.HighAvailabilityConf.HA_ZK_NAMESPACE
+import org.apache.kyuubi.ha.client.ServiceDiscovery.createServiceNode
+import org.apache.kyuubi.ha.client.ZooKeeperClientProvider.buildZookeeperClient
+import org.apache.kyuubi.util.ThreadUtils
+
+class ZooKeeperInstanceClient extends Logging {
```

Comment:
We have a zookeeper in package name already, and we use the client to interact with serviceDiscovery. Som How about 
```scala
class zookeeper.ServiceDiscoveryClient(serviceDiscovery: ServiceDiscovery) {
  def createClient.. // initialize
  def closeClient.. // stop
  // def preRegisterService
  def registerService.. // start
  // def postRegisterService
  // def preDeregisterService
  def deregisterService... // beforeStop, closeServiceNode
  def postDeregisterService ... // deletingChildrenIfNeeded
}
```

Reply:
or class ZookeeperServiceDiscoveryClient and remove zookeeper in package

Reply:
a typically calling order shall be
 `new ServiceDiscoveryClient` ->
 `createClient` ->
 `preRegisterService(if any)` -> `registerService` -> `postRegisterService (if any)`  ->
 `preDeregisterService` -> `deregisterService ` -> `postDeregisterService ` -> 
`closeClient`